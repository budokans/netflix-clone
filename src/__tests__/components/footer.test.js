import React from "react";
import { render } from "@testing-library/react";
import Footer from "../../components/footer";

describe("<Footer />", () => {
  it("renders the <Footer /> with populated data", () => {
    const { container, getByText } = render(
      <Footer>
        <Footer.Title>Questions? Phone 0800 472 310</Footer.Title>

        <Footer.Break />

        <Footer.Row>
          <Footer.Column>
            <Footer.Link href="#">FAQs</Footer.Link>
            <Footer.Link href="#">Investor Relations</Footer.Link>
            <Footer.Link href="#">Privacy</Footer.Link>
            <Footer.Link href="#">Speed test</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Help Center</Footer.Link>
            <Footer.Link href="#">Jobs</Footer.Link>
            <Footer.Link href="#">Cookie preferences</Footer.Link>
            <Footer.Link href="#">Legal notices</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Account</Footer.Link>
            <Footer.Link href="#">Ways to watch</Footer.Link>
            <Footer.Link href="#">Corporate Information</Footer.Link>
            <Footer.Link href="#">Netflix Originals</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Media Centre</Footer.Link>
            <Footer.Link href="#">Terms of Use</Footer.Link>
            <Footer.Link href="#">Contact us</Footer.Link>
          </Footer.Column>
        </Footer.Row>

        <Footer.Break />

        <Footer.Text>Netflix New Zealand</Footer.Text>
      </Footer>
    );

    expect(getByText("Questions? Phone 0800 472 310")).toBeTruthy();
    expect(getByText("FAQs")).toBeTruthy();
    expect(getByText("Help Center")).toBeTruthy();
    expect(getByText("Account")).toBeTruthy();
    expect(getByText("Media Centre")).toBeTruthy();
    expect(getByText("Netflix New Zealand")).toBeTruthy();

    const footerChildrenCount = container.firstChild.children.length;

    expect(footerChildrenCount).toBe(5);
    expect(container.firstChild).toMatchSnapshot();
  });
});
