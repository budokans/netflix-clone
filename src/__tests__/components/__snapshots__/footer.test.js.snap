// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders the <Footer /> with populated data 1`] = `
<div
  class="footer__Container-eh6b4z-0 qwEBr"
>
  <p
    class="footer__Title-eh6b4z-3 cSpvQd"
  >
    Questions? Phone 0800 472 310
  </p>
  <div
    class="footer__Break-eh6b4z-5 dJUYaW"
  />
  <div
    class="footer__Row-eh6b4z-2 ljTFhy"
  >
    <div
      class="footer__Column-eh6b4z-1 aKanr"
    >
      <a
        class="footer__Link-eh6b4z-4 bIcKKZ"
        href="#"
      >
        FAQs
      </a>
      <a
        class="footer__Link-eh6b4z-4 bIcKKZ"
        href="#"
      >
        Investor Relations
      </a>
      <a
        class="footer__Link-eh6b4z-4 bIcKKZ"
        href="#"
      >
        Privacy
      </a>
      <a
        class="footer__Link-eh6b4z-4 bIcKKZ"
        href="#"
      >
        Speed test
      </a>
    </div>
    <div
      class="footer__Column-eh6b4z-1 aKanr"
    >
      <a
        class="footer__Link-eh6b4z-4 bIcKKZ"
        href="#"
      >
        Help Center
      </a>
      <a
        class="footer__Link-eh6b4z-4 bIcKKZ"
        href="#"
      >
        Jobs
      </a>
      <a
        class="footer__Link-eh6b4z-4 bIcKKZ"
        href="#"
      >
        Cookie preferences
      </a>
      <a
        class="footer__Link-eh6b4z-4 bIcKKZ"
        href="#"
      >
        Legal notices
      </a>
    </div>
    <div
      class="footer__Column-eh6b4z-1 aKanr"
    >
      <a
        class="footer__Link-eh6b4z-4 bIcKKZ"
        href="#"
      >
        Account
      </a>
      <a
        class="footer__Link-eh6b4z-4 bIcKKZ"
        href="#"
      >
        Ways to watch
      </a>
      <a
        class="footer__Link-eh6b4z-4 bIcKKZ"
        href="#"
      >
        Corporate Information
      </a>
      <a
        class="footer__Link-eh6b4z-4 bIcKKZ"
        href="#"
      >
        Netflix Originals
      </a>
    </div>
    <div
      class="footer__Column-eh6b4z-1 aKanr"
    >
      <a
        class="footer__Link-eh6b4z-4 bIcKKZ"
        href="#"
      >
        Media Centre
      </a>
      <a
        class="footer__Link-eh6b4z-4 bIcKKZ"
        href="#"
      >
        Terms of Use
      </a>
      <a
        class="footer__Link-eh6b4z-4 bIcKKZ"
        href="#"
      >
        Contact us
      </a>
    </div>
  </div>
  <div
    class="footer__Break-eh6b4z-5 dJUYaW"
  />
  <p
    class="footer__Text-eh6b4z-6 jVcuq"
  >
    Netflix New Zealand
  </p>
</div>
`;
